### ISSUE TITLE
Parser Doesn't Handle Valid Pipe To "while"

### ISSUE THREAD

**Issue by tcf909** [BashSupport-Pro/bashsupport-pro#25]
```
eval "trap 'exit 129' INT TERM; ${CMD_ACTUAL}" 2>&1 | tee "${LOG_PATH}" while read -r LINE; do
            [[ "${PIPE_OUTPUT_CMD_LOG-}" == "true" ]] && LOG "CMD OUTPUT: ${LINE}"
        done
```
Results in a parsing error even though it is a valid bash.

(basically an output rewrite command)

BashPro latest.

**Comment by jansorg** [comment]
Thanks! I can confirm this with JetBrains Shell. I hope submit a patch to JetBrains to fix this for 2020.2.

**Comment by jansorg** [comment]
@tcf909 
Inserting a newline after the `|` and before the `tee` works around this bug in the parser. 
(I think the example you posted is missing a pipe `|` before the `while` keyword to make this valid Bash)

This should be working for you:
```
eval "trap 'exit 129' INT TERM; ${CMD_ACTUAL}" 2>&1 |
 tee "${LOG_PATH}" | while read -r LINE; do
            [[ "${PIPE_OUTPUT_CMD_LOG-}" == "true" ]] && LOG "CMD OUTPUT: ${LINE}"
        done
```

**Comment by tcf909** [comment]
> @tcf909
> Inserting a newline after the `|` and before the `tee` works around this bug in the parser.
> (I think the example you posted is missing a pipe `|` before the `while` keyword to make this valid Bash)
> 
> This should be working for you:
> 
> ```
> eval "trap 'exit 129' INT TERM; ${CMD_ACTUAL}" 2>&1 |
>  tee "${LOG_PATH}" | while read -r LINE; do
>             [[ "${PIPE_OUTPUT_CMD_LOG-}" == "true" ]] && LOG "CMD OUTPUT: ${LINE}"
>         done
> ```

@jansorg perfect. Thank you sir.

**Comment by jansorg** [comment]
Tracking at https://youtrack.jetbrains.com/issue/IDEA-243473

**Comment by jansorg** [comment]
This will be fixed by the the update, which will be out soon. The next version will not depend on the Shell plugin anymore, therefore it's possible to release this fix earlier.

**Comment by jansorg** [comment]
In the meanwhile the fix for this has been released for 2019.3 to 2020.2.
